usr/iscsiadm.c

```diff
@@ -3013,8 +3013,78 @@ static int exec_fw_disc_op(discovery_rec_t *drec, struct list_head *ifaces,
 	return rc;
 }
 
+static int fill_in_default_fw_values(node_rec_t *rec, struct list_head *params)
+{
+	struct user_param *param;
+	int rc;
+
+	/* handle compatability issues for iface.transport_name */
+	if (!list_empty(params)) {
+		rc = verify_node_params(params, rec);
+		if (rc)
+			return rc;
+	}
+
+	/* Must init this so we can check if user overrode them */
+	rec->session.initial_login_retry_max = -1;
+	rec->conn[0].timeo.noop_out_interval = -1;
+	rec->conn[0].timeo.noop_out_timeout = -1;
+	rec->session.scan = -1;
+
+	list_for_each_entry(param, params, list) {
+		/*
+		 * user may not have passed in all params that were set by
+		 * ibft/iscsi_boot, so clear out values that might conflict
+		 * with user overrides
+		 */
+		if (!strcmp(param->name, IFACE_NETNAME)) {
+			/* overriding netname so MAC will be for old netdev */
+			memset(rec->iface.hwaddress, 0,
+				sizeof(rec->iface.hwaddress));
+		} else if (!strcmp(param->name, IFACE_HWADDR)) {
+			/* overriding MAC so netdev will be for old MAC */
+			memset(rec->iface.netdev, 0, sizeof(rec->iface.netdev));
+		} else if (!strcmp(param->name, IFACE_TRANSPORTNAME)) {
+			/*
+			 * switching drivers so all old binding info is no
+			 * longer valid. Old values were either for offload
+			 * and we are switching to software or the reverse,
+			 * or switching types of cards (bnx2i to cxgb3i).
+			 */
+			memset(&rec->iface, 0, sizeof(rec->iface));
+			iface_setup_defaults(&rec->iface);
+		}
+	}
+
+	if (!list_empty(params)) {
+		rc = idbm_node_set_rec_from_param(params, rec, 0);
+		if (rc)
+			return rc;
+	}
+
+	/*
+	 * For root boot we could not change this in older versions so
+	 * if user did not override then use the defaults.
+	 *
+	 * Increase to account for boot using static setup.
+	 */
+	if (rec->session.initial_login_retry_max == -1)
+		rec->session.initial_login_retry_max = 30;
+
+	/* we used to not be able to answer so turn off */
```

Comment:
This was a iscsistart specific comment. When it was created we did nops in userspace so if we sent a nop off we probably would exit before we got a response. You would hit a bug like we fixed the other day. I think we also defaulted to nops off so that code in iscsistart is to keep that behavior (at some point the default was to turn them on and so we had to override in iscsistart).

For as long as we could do ibft/fw based boots we could do kernel based nops so you probably want to change the comment or drop it. Maybe just change the comment or add a mention in the git commit that we are turning off by default now to optimize for the single path case. If multipath users were relying on the defaults for fast boots during failures then they need to change them.